### PR TITLE
Draft of code splitting post

### PR DESCRIPTION
**Pull request by swannodette**
Work in progress

### CODE REVIEW

File: content/news/2017-07-14-code-splitting.adoc [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+ (:require [cljs.loader :as loader]
+           [goog.dom :as gdom]
+           [goog.events :as events])
+ (:import [goog.events EventType]))
+
+(events/listen (gdom/getElement "button") EventType.CLICK
+  (fn [e]
+    (loader/load :bar
+      (fn [e]
+        ((resolve 'bar.core/woz))))))
```
Comment (martinklepsch) [inline content/news/2017-07-14-code-splitting.adoc]: A real world example would further strengthen this example IMO. e.g.
``` 
(loader/load :admin
  (fn [e]
    ((resolve 'admin.core/initialize!))))))
```

File: content/news/2017-07-14-code-splitting.adoc [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+a set of possible module assignments. For example we find all dependencies of
+`foo.core` and we assume they will go into `:module-a` - even `cljs.core`, the
+standard library.
+
+But of course `:module-b` will also assign `cljs.core` to itself. So
+`cljs.core` module assignment is `[:module-a :module-b]`. However we can
+only choose one. To choose we first find all the common parent modules. Once
+found, we pick the module with the greatest `:depth` value.
+
+Finally, any orphans will be assigned to `:cljs-base`.
```
Comment (rauhs) [inline content/news/2017-07-14-code-splitting.adoc]: Suggestion: Add a note to use `:verbose` compiler option to see how the modules graph were computed.

File: content/news/2017-07-14-code-splitting.adoc [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+Readers familiar with Webpack will note that this is a simplification which
+doesn't require managing splits via source code or any additional plugins.
```
Comment (anmonteiro) [inline content/news/2017-07-14-code-splitting.adoc]: I don't know to what extent this is true, since the loader acts like the dynamic import in Webpack

Comment (swannodette) [inline content/news/2017-07-14-code-splitting.adoc]: @anmonteiro I'll clarify that I'm really only talking about the split definition, this requires more manual finagling via source + plugins.

File: content/news/2017-07-14-code-splitting.adoc (hunk 2 above)
Comment (swannodette) [inline content/news/2017-07-14-code-splitting.adoc]: @rauh good point

**Pull request merged**